app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -711,7 +711,7 @@ define([
                 }
             );
 
-            if (result.oldPrice.amount !== result.finalPrice.amount) {
+            if (typeof result != 'undefined' && result.oldPrice.amount !== result.finalPrice.amount) {
```

Comment:
typeof always returns a string so it's not necessary or standard practice to use strict comparison